src/com/unbounce/dogstatsd/ring.clj

```diff
@@ -0,0 +1,105 @@
+(ns com.unbounce.dogstatsd.ring
+  "Ring middleware for instrumenting HTTP request/responses.
+
+  The middleware provides the following dogstatsd metrics:
+
+  - http.1xx  - counter of 1xx responses
+  - http.2xx  - counter of 2xx responses
+  - http.3xx  - counter of 3xx responses
+  - http.4xx  - counter of 4xx responses
+  - http.5xx  - counter of 5xx responses
+  - http.count     - counter for total requests
+  - http.exception - counter for exceptions raised
+  - http.duration  - histogram of request duration
+
+  See `wrap-ring-metric` for more information.
+  "
+  (:require [com.unbounce.dogstatsd.core :as statsd])
+  (:import [java.util.concurrent ThreadLocalRandom]))
+
+(defn sample? [sample-rate]
+  (if sample-rate
+    (< (.nextFloat (ThreadLocalRandom/current)) sample-rate)
+    true))
+
+(defn healthcheck-request?
+  "Returns true if the healthcheck request is for a known health-check.
+
+  This function is aware of the following health-checks
+  - AWS ELB HealthChecks
+  - Pingdom
+  "
+  [request]
+  (let [user-agent (get-in request [:headers "user-agent"] "-")]
+    (re-matches #"(?i).*(ELB-HealthChecker|Pingdom).*" user-agent)))
```

Comment:
Is it worth making these configurable, with these defaults? Not sure if there are any other health check user-agents that a user of this middleware might want to exclude.

Reply:
It is configurable! In the middleware you can set a custom `loggable?` function that overrides this.